vm/qemu/qemu.go

```diff
@@ -534,7 +534,26 @@ func splitArgs(str, templateDir string, index int) (args []string) {
 		if arg == "" {
 			continue
 		}
-		arg = strings.ReplaceAll(arg, "{{INDEX}}", fmt.Sprint(index))
+		// 0-7 VFs/Device, if index >= 8, Device# += (index / 8)
+		if strings.Contains(arg, "{{INDEX}}") {
+			arg = strings.ReplaceAll(arg, "{{INDEX}}", fmt.Sprint(index%8))
+			if index > 7 && strings.Contains(arg, "vfio-pci,host=") {
+				// VFs are 0:7. Need to increment the Device if VF >= 8
+				args1 := strings.Split(arg, "vfio-pci,host=")
+				args2 := strings.Split(args1[1], ":")
```

Comment:
If args1 has length 1, args1[1] will panic.
The same for args2[1] below.

Reply:
Can you provide a suggested fix for these proposed changes (I am not a go expert)? Thank you.

Reply:
It should be enough to just check the length of the array after splitting (for the case the user provided invalid arguments).

E.g.
```
func handleVfioPciArg(arg string) string {
    args1 := strings.Split(arg, "vfio-pci,host=")
    if len(args1) != 2 {
        return arg
    }
    args2 := strings.Split(args1[1], ":")
    if len(args2) != 2 {
        return arg
    }
    // ...
}
```
Here we just abort replacement if we figure out it doesn't match our expectations. A somewhat even better approach would be to return an `error` in that case (from `splitArgs` as well -- `func splitArgs(str, templateDir string, index int) ([]string, error)`)

And note that at least for the first splitting you might need just https://pkg.go.dev/strings@go1.16.15#HasPrefix and https://pkg.go.dev/strings@go1.16.15#TrimPrefix

Reply:
Requested change made - see: https://github.com/gkennedy12/syzkaller/tree/pull_072622_1